src/main/java/com/google/sps/servlets/DeleteMarker.java

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that process information about user's markers. */
+@SuppressWarnings("serial")
+@WebServlet("/delete_marker")
+public class DeleteMarker extends HttpServlet {
+  /** Processes POST request by deleting marker by received id. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get Id from the request.
+    try {
+      String contentId = request.getParameter("contentId");
+      Long id = Long.parseLong(contentId);
+
+      Key markerKey = KeyFactory.createKey("Marker", id);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.delete(markerKey);
+    } catch (Exception e) {
```

Comment:
It is better to catch more specific exceptions, and handle each exception type in a way that is appropriate for the type. See https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/DatastoreService.html#delete for the exceptions it will throw.

It will throw an IllegalArgumentException if the Key is invalid, so you can send that back to the client with  a specific message. You probalby don't have to log it.

If a DatastoreFailureException happens, you could send a message like "Unexpected error while deleting the marker." to the client, and log the exception details.

`catch (Exception e)` should be the last resort to catch all unexpected errors and just log the exception details.

As with the other comment about logging, please create and issue and add a TODO here to fix it in a follow-up PR.

Reply:
I added it to the issue (№ 14).